api/src/org/apache/cloudstack/api/command/admin/host/ListHostsCmd.java

```diff
@@ -203,9 +201,22 @@ public void execute() {
                 hostResponse.setObjectName("host");
                 hostResponses.add(hostResponse);
             }
-
             response.setResponses(hostResponses, result.second());
         }
+        // Remove sensitive details from host response
+        List<HostResponse> hostResponsesList = response.getResponses();
+        for (HostResponse hostResponse: hostResponsesList) {
+            Map<String, String> hostDetails = hostResponse.getDetails();
+            if (hostDetails == null) continue;
+            if (hostDetails.containsKey("username")) {
+                hostDetails.remove("username");
+            }
+            if (hostDetails.containsKey("password")) {
+                hostDetails.remove("password");
+            }
+            hostResponse.setDetails(hostDetails);
```

Comment:
Why not perform this filtering on a defensive copy of the map in HostResponse.setDetails()?  I am also concerned about directly manipulating the details map in this method as it changes the view for any other objects holding references to it.